DIPs/DIP1009.md

```diff
@@ -42,38 +44,51 @@ body
     ulong ret = 0;
 }
 ```
-A second reason contracts are hard to use is that they require the use of the keyword `do` (or `body`, see [DIP1003](https://github.com/dlang/DIPs/blob/master/DIPs/DIP1003.md)) before the function body. Any way of writing them that could eliminate this need would be an improvement.
 
-Finally, `in` and `out` contracts in D require explicitly writing `assert` statements, or some equivalent functionality, in their code block in order to perform checks. But because of the nature of contracts, `assert` can be assumed in the vast majority of cases, and it is tedious to have to write it specifically. For the common case, the mere presence of `in` or `out` is sufficient. For those programmers who require the use of other forms of checking, the existing syntax will remain as is. 
+2. Contracts require the use of the keyword `do` (or `body`, see [DIP1003](https://github.com/dlang/DIPs/blob/master/DIPs/DIP1003.md)) before the function body. Any way of writing them that eliminated this need would be an improvement.
+
+3. `in` and `out` contracts in D require explicitly writing `assert` statements, or some equivalent, in their code block in order to perform checks. But because of the nature of contracts, `assert` can be assumed in the vast majority of cases, and it is tedious to have to write it specifically. For the common case, the mere presence of `in` or `out` is sufficient. For those programmers who require the use of other forms of checking, the existing syntax will remain unchanged. 
```

Comment:
"`in` and `out` contracts in D require explicitly writing `assert` statements, or some equivalent, in their code block in order to perform checks. But because of the nature of contracts, `assert` can be assumed in the vast majority of cases, and it is tedious to have to write it specifically. For the common case, the mere presence of `in` or `out` is sufficient."

`assert` statements, are some equivalent, are used to perform checks within `in` and `out` contract blocks. Currently, the language requires these statements to be explicitly written. This is tedious and, for the common case, unnecessary. The presence of a single Boolean expression in an `in` or `out` block should be sufficient to assume an implicit `assert`.